src/core/singleton_decorator.py

```diff
@@ -0,0 +1,13 @@
+# structure for the singleton decorator
+
+
+def singleton(cls):
+
+    instance = [None]
```

Comment:
Why are you using `None` for this array?

Reply:
It is to check in the code below if the instance[0] is None, thus i know that i'm creating the instance for the firts time.
it is the same to put instance[0] = None